autorest/preparer_test.go

```diff
@@ -437,11 +438,76 @@ func TestWithFormDataSetsContentLength(t *testing.T) {
 		t.Fatalf("autorest: WithFormData failed with error (%v)", err)
 	}
 
+	expected := "age=42&name=Rob+Pike"
```

Comment:
I believe it is the case that iteration ordering is intentionally randomized to prevent people from taking a dependency on it.

https://blog.golang.org/go-maps-in-action Ctrl-F "Since Go 1"...